### PR TITLE
chore: add int tests to verify permissions of AllDataReadWrite tokens

### PR DESCRIPTION
**Pull request by cprice404**
This commit adds a token-scope shared integration test, which
creates a new AllDataReadWrite token, and then creates a cache
client using that token.  Then we make some positive and negative
assertions to validate that the client has access to the operations
we expect, and gets rejected from other ones.

### CODE REVIEW

**Review by rlinehan — APPROVED** [review]

**Comment by cprice404** [comment]
rebased

File: packages/common-integration-tests/src/token-scope.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      allDataReadWriteClient = cacheClientFactory(allDataReadWriteToken);
+    });
+    it('cannot create a cache', async () => {
+      const createCacheResponse = await allDataReadWriteClient.createCache(
+        'FOOFOOFOO'
+      );
+      expect(createCacheResponse).toBeInstanceOf(CreateCache.Error);
+      const createCacheError = createCacheResponse as CreateCache.Error;
+      expect(createCacheError.errorCode()).toEqual(
+        MomentoErrorCode.AUTHENTICATION_ERROR
```
Comment (rlinehan) [inline packages/common-integration-tests/src/token-scope.ts:42]: I just merged the change that changes the grpc status from `UNAUTHENTICATED` to `PERMISSION_DENIED`

Comment (cprice404) [reply packages/common-integration-tests/src/token-scope.ts:42]: okay well we can fix that on a subsequent PR when we see it fail?

Comment (rlinehan) [reply packages/common-integration-tests/src/token-scope.ts:42]: sure I just think it might fail as soon as you merge this in to main 😅

**Review by rlinehan — APPROVED** [review]

**Pull request merged**